docs/proposals/fluentd-limit-requests-to-apiserver.md

```diff
@@ -0,0 +1,63 @@
+# Modifications to Fluentd Metadata Retrieval Proposal
+
+##Motivation
+In the current design (v3.4) of the EFK stack, there is a Fluentd pod deployed to each Openshift node. Fluentd is responsible for reading container logs, adding pod metadata (e.g. labels, namespace) to individual log entries, and sending the logs to the Elasticsearch cluster. The pod metadata is sourced by watching the Openshift API server for Pod information via the [Kubernetes Metadata Filter](https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter)[2]. Testing at larger scales of Openshift clusters has shown issues where Fluentd is unable to connect to the API server because it is limited by the **maxRequestsInFlight** configuration setting [1].  This results in logs not being shipped to the aggregator and thus being unavailable.
```

Comment:
Can we add the specific reason we need the metadata retrieval in the first place?

In a pod, the fluentd process has access to the `pod_name` and the `namespace_name`, but is missing the:
 * `pod_id` (or `metadata.uid` from k8s)
 * `host` (or `spec.nodeName`)
 * `labels` (or `metadata.labels`)
 * `annotations` (or `metadata.annotations`)

The fluentd process currently needs to fetch the metadata associated with the pod/namespace and then cache it to allow it to scale.

The current problem comes from the fact that the API server connection appears to be kept open the entire time, not just when an API call needs to be made.